### PR TITLE
PLANET-4827 Add page to import archive files

### PR DESCRIPTION
**Pull request by Inwerpsel**
This should probably be in master-theme instead, but started off here
and it takes a while to move this to master-theme as the it doesn't have
the same controllers. So pushing this one to have it on the server
already.

### CODE REVIEW

File: classes/controller/menu/class-archive-import.php [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+		$items = $data->item;
+
+		$total    = count( $items );
+		$errors   = [];
+		$existing = [];
+		$created  = [];
+
+		foreach ( $items as $item ) {
+			$url = self::WAY_BACK_URL . $item->url;
+			if ( $this->does_post_exist( $url, $urls_imported_as_p4_content ) ) {
```
Comment (sagarsdeshmukh) [inline classes/controller/menu/class-archive-import.php:109]: Looks like the duplicate posts(previously imported) are not filtering/skipping with this check -
https://k8s.p4.greenpeace.org/international/?s=jailed+in+Russia+to+take+case+to+European+Court&orderby=_score
https://k8s.p4.greenpeace.org/international/?s=6+ways+corporate+lawsuits+kill+free+speech&orderby=_score
https://k8s.p4.greenpeace.org/international/?s=Global+movement+unites+against+Norwegian+oil+in+the+Arctic&orderby=_score

File: classes/controller/menu/class-archive-import.php [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+/**
+ * An importer for archived content from an XML file.
+ *
+ * @package P4GBKS\Controllers\Menu
+ */
+class Archive_Import extends Controller {
+
+	private const WAY_BACK_URL = 'https://wayback.archive-it.org/9650/';
+
+	private const P3_BASE_URL = 'http://p3-raw.greenpeace.org/';
```
Comment (sagarsdeshmukh) [inline classes/controller/menu/class-archive-import.php:21]: The fixed p3 link has a excuse for NL(https://github.com/greenpeace/planet4-gpi-export/blob/master/P3%20Archive%20Elastic%20Search%20data/data/netherlands/netherlands-nl_post_list_with_date_final-clean.csv)
We need to handle it with some hardcoded condition.

File: classes/controller/menu/class-archive-import.php [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+					'numberposts' => -1,
+				]
+			);
+			foreach ( $archives as $archive ) {
+				wp_delete_post( $archive->ID );
+			}
+			return 'Deleted all archived posts';
+		}
+
+		if ( isset( $_GET['done'] ) ) { //phpcs:ignore WordPress.Security.NonceVerification.Recommended
```
Comment (sagarsdeshmukh) [inline classes/controller/menu/class-archive-import.php:83]: I am not sure, when this condition executes? is it a part of XML import?

File: classes/controller/menu/class-archive-import.php [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+			if ( is_wp_error( $result ) ) {
+				$errors[] = $result;
+			} else {
+				$created[] = $result;
+			}
+		}
+
+		//phpcs:disable
+		echo 'Peak memory usage: ' . $this->readable_bytes( memory_get_peak_usage() );
+		echo "Total: $total<br/>";
```
Comment (sagarsdeshmukh) [inline classes/controller/menu/class-archive-import.php:152]: for consistency, could you please use single quotes for string.

File: classes/controller/menu/class-archive-import.php (hunk 3 above)
Comment (Inwerpsel) [reply classes/controller/menu/class-archive-import.php:83]: Ah you're right, this isn't used anymore, will remove it.

File: classes/controller/menu/class-archive-import.php (hunk 2 above)
Comment (Inwerpsel) [reply classes/controller/menu/class-archive-import.php:21]: Are you ok with taking this up later? It's only for a couple of NROs, so we can start with those where this is not the case.

File: classes/controller/menu/class-archive-import.php (hunk 4 above)
Comment (Inwerpsel) [reply classes/controller/menu/class-archive-import.php:152]: This string is using interpolation so I can't replace it with single quotes.

**Comment by sagarsdeshmukh** [comment]
Few minor comments, other than that PR looks good to me 👍

File: classes/controller/menu/class-archive-import.php (hunk 1 above)
Comment (Inwerpsel) [reply classes/controller/menu/class-archive-import.php:105]: Could be fixed but we might have to follow up on this

**Pull request merged**